sdk/tables/azure-data-tables/azure/data/tables/aio/_policies_async.py

```diff
@@ -3,317 +3,153 @@
-        ) or request.context.options.pop("data_stream_total", None)
-        download_stream_current = request.context.get(
-            "download_stream_current"
-        ) or request.context.options.pop("download_stream_current", None)
-        upload_stream_current = request.context.get(
-            "upload_stream_current"
-        ) or request.context.options.pop("upload_stream_current", None)
-        response_callback = request.context.get(
-            "response_callback"
-        ) or request.context.options.pop("raw_response_hook", self._response_callback)
+    :keyword float retry_backoff_factor: A backoff factor to apply between attempts after the second try
+     (most errors are resolved immediately by a second try without a delay).
+     In fixed mode, retry policy will alwasy sleep for {backoff factor}.
+     In 'exponential' mode, retry policy will sleep for: `{backoff factor} * (2 ** ({number of total retries} - 1))`
+     seconds. If the backoff_factor is 0.1, then the retry will sleep
+     for [0.0s, 0.2s, 0.4s, ...] between retries. The default value is 0.8.
 
-        response = await self.next.send(request)
-        await response.http_response.load_body()
+    :keyword int retry_backoff_max: The maximum back off time. Default value is 120 seconds (2 minutes).
 
-        will_retry = is_retry(response, request.context.options.get("mode"))
-        if not will_retry and download_stream_current is not None:
-            download_stream_current += int(
-                response.http_response.headers.get("Content-Length", 0)
-            )
-            if data_stream_total is None:
-                content_range = response.http_response.headers.get("Content-Range")
-                if content_range:
-                    data_stream_total = int(
-                        content_range.split(" ", 1)[1].split("/", 1)[1]
-                    )
-                else:
-                    data_stream_total = download_stream_current
-        elif not will_retry and upload_stream_current is not None:
-            upload_stream_current += int(
-                response.http_request.headers.get("Content-Length", 0)
-            )
-        for pipeline_obj in [request, response]:
-            pipeline_obj.context["data_stream_total"] = data_stream_total
-            pipeline_obj.context["download_stream_current"] = download_stream_current
-            pipeline_obj.context["upload_stream_current"] = upload_stream_current
-        if response_callback:
-            if asyncio.iscoroutine(response_callback):
-                await response_callback(response)
-            else:
-                response_callback(response)
-            request.context["response_callback"] = response_callback
-        return response
+    :keyword RetryMode retry_mode: Fixed or exponential delay between attemps, default is exponential.
 
+    :keyword int timeout: Timeout setting for the operation in seconds, default is 604800s (7 days).
+    """
```

Comment:
Just for context, why is the timeout setting so high?

Reply:
hmm good question - but it's a question for azure-core - as I just copying the docstring from the base retry policy (which is what this retry policy inherits from).